support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/validator/token/OAuth20RevocationRequestValidator.java

```diff
@@ -0,0 +1,52 @@
+package org.apereo.cas.support.oauth.validator.token;
+
+import org.apereo.cas.services.ServicesManager;
+import org.apereo.cas.support.oauth.OAuth20Constants;
+import org.apereo.cas.support.oauth.util.OAuth20Utils;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.JEEContext;
+import org.springframework.core.Ordered;
+
+/**
+ * This is {@link OAuth20RevocationRequestValidator}.
+ *
+ * @author Julien Huon
+ * @since 6.2.0
+ */
+@RequiredArgsConstructor
+@Slf4j
+@Getter
+@Setter
+public class OAuth20RevocationRequestValidator implements OAuth20TokenRequestValidator {
+    private final ServicesManager servicesManager;
+
+    private int order = Ordered.LOWEST_PRECEDENCE;
+
+    @Override
+    public boolean validate(final JEEContext context) {
+        val clientId = OAuth20Utils.getClientIdAndClientSecret(context).getLeft();
+        val registeredService = OAuth20Utils.getRegisteredOAuthServiceByClientId(this.servicesManager, clientId);
+
+        if (registeredService == null) {
+            LOGGER.warn("Provided client id [{}] cannot be matched against a service definition", clientId);
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean supports(final JEEContext context) {
+        val token = context.getRequestParameter(OAuth20Constants.TOKEN)
+            .map(String::valueOf).orElse(StringUtils.EMPTY);
+        val clientId = OAuth20Utils.getClientIdAndClientSecret(context).getLeft();
```

Comment:
If the token is blank you don't need to check/fetch the client-id. Return false immediately, and marginally help with performance.

Reply:
Done.